modin/experimental/core/execution/native/implementations/hdk_on_native/hdk_worker.py

```diff
@@ -39,8 +40,11 @@ def setup_engine(cls):
         Do nothing if it is initiliazed already.
         """
         if cls._executor is None:
+            cls._preffered_device = (
+                "CPU" if bool(HdkLaunchParameters.get()["cpu_only"]) else "GPU"
+            )
             cls._config = pyhdk.buildConfig(**HdkLaunchParameters.get())
-            cls._storage = pyhdk.storage.ArrowStorage(1)
+            cls._storage = pyhdk.storage.ArrowStorage(1, cls._config)
```

Comment:
Why did we not pass the config before?

Reply:
No idea, our API expects a config. Maybe @AndreyPavlenko knows?

Reply:
I suppose, it was not required for the previous implementation.
Btw, this code will be removed when #6286 is meged. The new Implementation will use the HDK class which also passes the config to ArrowStorage() - https://github.com/intel-ai/hdk/blob/main/python/pyhdk/hdk.py#L1879